addons/account_analytic_default_purchase/models/purchase_order_line.py

```diff
@@ -7,15 +7,29 @@
 class PurchaseOrderLine(models.Model):
     _inherit = "purchase.order.line"
 
-    @api.onchange('product_id', 'date_order')
-    def _onchange_product_id_date(self):
-        default_analytic_account = self.env['account.analytic.default'].sudo().account_get(
-            product_id=self.product_id.id,
-            partner_id=self.order_id.partner_id.id,
-            user_id=self.env.uid,
-            date=self.date_order,
-            company_id=self.company_id.id,
-        )
-        if default_analytic_account:
-            self.account_analytic_id = default_analytic_account.analytic_id.id
-            self.analytic_tag_ids = [(6, 0, default_analytic_account.analytic_tag_ids.ids)]
+    account_analytic_id = fields.Many2one(compute='_compute_analytic_id', store=True, readonly=False)
+    analytic_tag_ids = fields.Many2many(compute='_compute_tag_ids', store=True, readonly=False)
+
+    @api.depends('product_id', 'date_order')
+    def _compute_analytic_id(self):
+        for rec in self:
+            default_analytic_account = rec.env['account.analytic.default'].sudo().account_get(
+                product_id=rec.product_id.id,
+                partner_id=rec.order_id.partner_id.id,
+                user_id=rec.env.uid,
+                date=rec.date_order,
+                company_id=rec.company_id.id,
+            )
+            rec.account_analytic_id = rec.account_analytic_id or default_analytic_account.analytic_id
```

Comment:
No need to do that complex search with `account_get` at all if you know that you already have a value and you won't override it